src/WebJobs.Script/Description/DotNet/FunctionAssemblyLoadContext.cs

```diff
@@ -39,6 +39,8 @@ public FunctionAssemblyLoadContext(string basePath)
                 throw new ArgumentNullException(nameof(basePath));
             }
 
+            _runtimeAssemblies = new Lazy<Dictionary<string, ScriptRuntimeAssembly>>(DependencyHelper.GetRuntimeAssemblies);
```

Comment:
Reloading this will affect cold start, I assume? Probably a premature optimization -- but you could check to see if the compat mode had changed at all since last load. I'd assume most of the time this won't even need to be reloaded. But I guess we can punt that until we see if this is an issue or not.

Reply:
Yes, I can make that change. The impact shouldn't be super significant, which is why I opted for the cleaner/less complex approach

Reply:
It's okay to leave it as-is -- was just trying to think through the differences now. We can always address it later if needed.